### PR TITLE
feat(cli): add `set default-env` command

### PR DESCRIPTION
**Pull request by edvald**
Locally set your default environment for a project. Simple quality of life improvement, that I'm sure many will enjoy (I for sure will).

cc @jalper :)

### CODE REVIEW

**Comment by edvald** [comment]
E2E test failures seem to do with the GitHub certificate issue, not the PR. Perhaps @stefreak or @twelvemo could look into that on Monday?

File: core/src/cli/cli.ts [hunk 1]
```diff
@@ -239,7 +237,7 @@ ${renderCommands(commands)}
     processRecord?: GardenProcess
     workingDir: string
   }) {
-    const {
+    let {
```
Comment (vvagaytsev) [inline core/src/cli/cli.ts:240]: It looks like these variables are not reassigned later in the code. Should we keep the `const` declaration here?

**Review by vvagaytsev — APPROVED** [review]
Looks good, thank you! 👍 
I've left a minor non-blocking comment.

**Comment by twelvemo** [comment]
> E2E test failures seem to do with the GitHub certificate issue, not the PR. Perhaps @stefreak or @twelvemo could look into that on Monday?

Seems to have been resolved in the mean time. I re-tried the jobs and they succeeded. Maybe CircleCi updated `known_hosts` on their runners.

**Comment by vvagaytsev** [comment]
I've fixed the tests and resolved the conflicts. Let's wait for the CI pipeline completion to see if everything is ok.

**Comment by vvagaytsev** [comment]
Ah, there are more conflicts introduced by #4002.

**Comment by vvagaytsev** [comment]
@edvald @eysi09 could you check the conflicts, please? I'm a bit confused with the namespace and environment names there.

**Comment by eysi09** [comment]
@vvagaytsev, just resolved (via GitHub). 

@edvald, previously we used to log a similar string, except it was `<namespace>.<environment>`. We changed that to the current format because we felt the former was intuitive. 

Now we're changing it to `<environment>.<namespace>`. Just wanted to confirm that that's intentional and that from now on we'll referring to the environment + namespace pair with that format? 

If so, we need to update the docs as well but we can do that separately, I can add an item to #3961.

**Comment by vvagaytsev** [comment]
@eysi09 now there are some build issues

**Comment by eysi09** [comment]
Roger that, will fix. That's what I get for trying to do this in the GitHub UI.

**Review by vvagaytsev — APPROVED** [review]
LGTM! 🚀 Let's squash the commits before merging.

**Comment by vvagaytsev** [comment]
Hmm... For some reason GH says there are still some conflicts.

**Comment by vvagaytsev** [comment]
Ah, there was a merge commit instead of rebasing on top of `0.13`. This can be merged with a merge commit.

**Comment by edvald** [comment]
> @vvagaytsev, just resolved (via GitHub).
> 
> @edvald, previously we used to log a similar string, except it was `<namespace>.<environment>`. We changed that to the current format because we felt the former was intuitive.
> 
> Now we're changing it to `<environment>.<namespace>`. Just wanted to confirm that that's intentional and that from now on we'll referring to the environment + namespace pair with that format?
> 
> If so, we need to update the docs as well but we can do that separately, I can add an item to #3961.

It was intentional, yes. To me it felt most logical to use the same format as when you specify the `--env` flag, e.g. `--env=dev.something`. This is also in the spirit of reducing references to "namespace" outside of actual Kubernetes namespaces.

**Pull request merged**